worlds/yoshisisland/Regions.py

```diff
@@ -0,0 +1,225 @@
+        create_region(multiworld, player, locations_per_region, location_cache, "6-7"),
+        create_region(multiworld, player, locations_per_region, location_cache, "6-8"),
+        create_region(multiworld, player, locations_per_region, location_cache, "Bowser's Room"),
+
+    ]
+    if world.options.extras_enabled.value == 1:
+        regions.insert(68, create_region(multiworld, player, locations_per_region, location_cache, "6-Extra"))
+        regions.insert(58, create_region(multiworld, player, locations_per_region, location_cache, "5-Extra"))
+        regions.insert(48, create_region(multiworld, player, locations_per_region, location_cache, "4-Extra"))
+        regions.insert(38, create_region(multiworld, player, locations_per_region, location_cache, "3-Extra"))
+        regions.insert(28, create_region(multiworld, player, locations_per_region, location_cache, "2-Extra"))
+        regions.insert(18, create_region(multiworld, player, locations_per_region, location_cache, "1-Extra"))
+
+    if world.options.minigame_checks.value >= 2:
+        regions.insert(74, create_region(multiworld, player, locations_per_region, location_cache, "6-Bonus"))
+        regions.insert(63, create_region(multiworld, player, locations_per_region, location_cache, "5-Bonus"))
+        regions.insert(52, create_region(multiworld, player, locations_per_region, location_cache, "4-Bonus"))
+        regions.insert(41, create_region(multiworld, player, locations_per_region, location_cache, "3-Bonus"))
+        regions.insert(29, create_region(multiworld, player, locations_per_region, location_cache, "2-Bonus"))
+        regions.insert(19, create_region(multiworld, player, locations_per_region, location_cache, "1-Bonus"))
+    multiworld.regions += regions
+
+    connectStartingRegion(multiworld, player, world)
+
+    bosses = BossReqs(player, world)
+
+    multiworld.get_region('Overworld', player).add_exits(['World 1', 'World 2', 'World 3', 'World 4', 'World 5', 'World 6'],
+                                                        {'World 1': lambda state: state.has('World 1 Gate', player),
+                                                        'World 2': lambda state: state.has('World 2 Gate', player),
+                                                        'World 3': lambda state: state.has('World 3 Gate', player),
+                                                        'World 4': lambda state: state.has('World 4 Gate', player),
+                                                        'World 5': lambda state: state.has('World 5 Gate', player),
+                                                        'World 6': lambda state: state.has('World 6 Gate', player)})
+    cur_world = 1
+    cur_level = 1
+    for i in range(47):
+        multiworld.get_region(f'World {cur_world}', player).add_exits([world.level_location_list[i]])
+        if cur_level > 7:
+            cur_level = 1
+            cur_world += 1
+        else: cur_level += 1

```

Comment:
```suggestion
    for cur_world in range(1, 7):
        for cur_level in range(7):
            multiworld.get_region(f"World {cur_world}", player).add_exits([world.level_location_list[cur_level]])
```

Reply:
maybe i'm too tired to touch this up currently but this does not work. I see what you're going for though